config/routes.rb

```diff
@@ -0,0 +1,8 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  root 'tasks#index'
+  resources :tasks
+
+  patch 'tasks/:id/complete', to: 'tasks#mark_complete', as: 'task_complete'
```

Comment:
Good work getting `resources` figured out. You can also add non-restful routes by adding a block to the call to resources:

```ruby
resources :tasks do
  patch '/complete', to: 'tasks#mark_complete', as: 'task_complete'
end
```